pkg/noderesourcetopology/cache/store.go

```diff
@@ -197,38 +198,57 @@ func (cnt counter) Len() int {
 }
 
 // podFingerprintForNodeTopology extracts without recomputing the pods fingerprint from
-// the provided Node Resource Topology object.
-func podFingerprintForNodeTopology(nrt *topologyv1alpha2.NodeResourceTopology) string {
+// the provided Node Resource Topology object. Returns the expected fingerprint and the method to compute it.
+func podFingerprintForNodeTopology(nrt *topologyv1alpha2.NodeResourceTopology, method apiconfig.CacheResyncMethod) (string, apiconfig.CacheResyncMethod) {
```

Comment:
maybe change signature to:
```go
func podFingerprintForNodeTopology(nrt *topologyv1alpha2.NodeResourceTopology, method apiconfig.CacheResyncMethod) (string, bool) {
```
and the second return param will be `wantsOnlyExclRes`.
Then when passing to `checkPodFingerprintForNode` we can pass only boolean and not `pfpMethod`

Reply:
good point. That was the original version of the code. I'm still torn between YAGNI and keep the signature simple (and don't leak `apiconfig` too much all around the codebase) and be generic and forward looking.
Now that you hihglight this, I'm leaning towards reverting to YAGNI and implement your suggestion